odoo/tools/mail.py

```diff
@@ -230,26 +309,31 @@ def html_normalize(src, filter_callback=None):
     src = re.sub(r'</?o:.*?>', '', src)
 
     try:
-        doc = html.fromstring(src)
+        doc, signle_body_element = fromstring(src)
     except etree.ParserError as e:
         # HTML comment only string, whitespace only..
         if 'empty' in str(e):
             return ""
         raise
 
     # perform quote detection before cleaning and class removal
-    if doc is not None:
-        for el in doc.iter(tag=etree.Element):
-            tag_quote(el)
+    for el in doc.iter(tag=etree.Element):
+        tag_quote(el)
 
     if filter_callback:
         doc = filter_callback(doc)
 
     src = html.tostring(doc, encoding='unicode')
 
-    # this is ugly, but lxml/etree tostring want to put everything in a
-    # 'div' that breaks the editor -> remove that
-    if src.startswith('<div>') and src.endswith('</div>'):
+    if not signle_body_element and doc.tag == 'div':
```

Comment:
according to my knowledge, it is impossible to have `<div ...>` here. But we still can add an extra condition or replace the current one by using ` and src.startswith('<div>`)`  for safty